test/integration/plugins/storpool/TestStorPoolTiers.py

```diff
@@ -0,0 +1,554 @@
+
+        cls.template_name = storpool_primary_storage.get("name")
+
+        storage_pool = list_storage_pools(
+            cls.apiclient,
+            name=cls.template_name
+        )
+
+        service_offerings = list_service_offering(
+            cls.apiclient,
+            name=cls.template_name
+        )
+
+        disk_offerings = list_disk_offering(
+            cls.apiclient,
+            name="ssd"
+        )
+
+        if storage_pool is None:
+            storage_pool = StoragePool.create(cls.apiclient, storpool_primary_storage)
+        else:
+            storage_pool = storage_pool[0]
+        cls.storage_pool = storage_pool
+        cls.debug(pprint.pformat(storage_pool))
+        if service_offerings is None:
+            service_offerings = ServiceOffering.create(cls.apiclient, storpool_service_offerings)
+        else:
+            service_offerings = service_offerings[0]
+        # The version of CentOS has to be supported
+        template = get_template(
+            cls.apiclient,
+            cls.zone.id,
+            account="system"
+        )
+
+        if template == FAILED:
+            assert False, "get_template() failed to return template\
+                    with description %s" % cls.services["ostype"]
+
+        cls.services["domainid"] = cls.domain.id
+        cls.services["small"]["zoneid"] = cls.zone.id
+        cls.services["templates"]["ostypeid"] = template.ostypeid
+        cls.services["zoneid"] = cls.zone.id
+
+        cls.service_offering = service_offerings
+        cls.debug(pprint.pformat(cls.service_offering))
+
+        cls.template = template
+        cls.random_data_0 = random_gen(size=100)
+        cls.test_dir = "/tmp"
+        cls.random_data = "random.data"
+        return
+
+    @classmethod
+    def getDiskOffering(cls, dataDiskOffering, qos, resValue):
+        disk_offerings = list_disk_offering(cls.apiclient, name=dataDiskOffering.get("name"))
+        if disk_offerings is None:
+            disk_offerings = DiskOffering.create(cls.apiclient, services=dataDiskOffering, custom=True)
+            cls.resourceDetails(qos, disk_offerings.id, resValue)
+        else:
+            disk_offerings = disk_offerings[0]
+            cls.resourceDetails(qos, disk_offerings.id, )
+        return disk_offerings
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.cleanUpCloudStack()
+
+    @classmethod
+    def cleanUpCloudStack(cls):
+        try:
+            # Cleanup resources used
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+
+        if self.unsupportedHypervisor:
+            self.skipTest("Skipping test because unsupported hypervisor\
+                    %s" % self.hypervisor)
+        return
```

Comment:
```suggestion
        self.cleanup = []
        return
```

Reply:
as I mentioned in one of the suggestions, I don't add all objects to be cleaned. Everything created under the account created from the tests is destroyed at the end